.github/workflows/default.yaml

```
@@ -120,3 +120,34 @@ jobs:
         with:
           name: build-logs-${{ runner.os }}
           path: btlayer-*.log
+
+  CodeQL-tests:
+
+    runs-on: [ ubuntu-22.04 ]
+
+    permissions:
+      security-events: write
+
+    steps:
+
+      - name: Install Squid prerequisite Linux packages
+        if: runner.os == 'Linux'
+        run: |
+          # required for "apt-get build-dep" to work
+          sudo sed --in-place -E 's/# (deb-src.*updates main)/  \1/g' /etc/apt/sources.list
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 build-dep squid
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+
+      - name: Build Squid
+        run: |
+          ./bootstrap.sh
+          ./test-suite/buildtest.sh check ./test-suite/buildtests/layer-02-maximus.opts
```

Our test-builds.sh script performs a few actions before running the equivalent of the above command. If possible, let's reuse that higher-level script instead of calling buildtest.sh directly:

```suggestion
          ./test-builds.sh ./test-suite/buildtests/layer-02-maximus.opts
```

As an added bonus, we would not need to bootstrap manually in this case because the higher-level script will do that for us.